src/gz/read.rs

```diff
@@ -90,13 +90,11 @@ impl<R: Read + Write> Write for GzEncoder<R> {
     }
 }
 
-/// A gzip streaming decoder
+/// A decoder for a single member of a gzip file. Prefer [MultiGzDecoder] for
+/// most uses.
```

Comment:
```suggestion
/// A decoder for a gzip file with a single member.
```

I'd agree that we shouldn't *just* present this as a decoder for gzip in general, and should capture the distinction in the summary, but I'd stop short of telling people to *prefer* MultiGzDecoder.